src/content/Blog/running-vllm-on-akash/index.md

```diff
@@ -0,0 +1,143 @@
+
+Steps to Troubleshoot
+1. Read the logs for the vllm container and the UI container
+2. Check that the model weights are downloading correctly and can be loaded into VRAM 
+3. Check the model size is smaller than VRAM and container disk space 
+4.  Check the huggingface model repo to make sure the repo and author are correct
+5. Check the huggingface repo model type and make sure its not GGUF and is compatible with VLLM (GPTQ, AWQ, GGML, Safetensors etc)
+6. Make sure that the model is [officially supported by vLLM](https://docs.vllm.ai/en/latest/models/supported_models.html)
+7. Check the environment variables for setting the `VLLM_API_KEY` and `OPENAI_API_KEYS` environment variables match 
+8.  Check that the the `HUGGING_FACE_HUB_TOKEN` is set for downloading models
+9. if using the api externally make sure you have updated the url to use the deployer endpoint variable which can be done by setting `OPENAI_API_BASE`
+10. if you have checked all of these and still have problems than open a issue in the awesome-akash repo and tag @rakataprime.  In the issue,  please provide your logs and deployment used with the huggingface token and other secrets set to XXXXXXXXXXXX. 
+
+## Choosing The Right GPU for An LLM Model
+
+- VLLM supported files formats include : GPTQ, AWQ, GGML(squeezelm ), and pytorch pth/bin and safetensors files
+- VLLM DOES NOT SUPPORT GGUF files
+- Check to make sure the base foundation model is supported [here](https://docs.vllm.ai/en/latest/models/supported_models.html)
+- Make sure that the file size is smaller than the VRAM of the GPU with a little buffer room.  Try to make sure the model files fit within 90% of the Total VRAM.
+- Make sure the vllm container has a large of disk space to store the model weights.
+- If you are doing something funny with really large context lengths you can use the tools below to help estimate VRAM utilization.
+- Remember that really large models like Grok sometimes require multiple gpus.  the max number of gpus supported by this vllm example is 8 gpus or 640Gb VRAM for a100
+- If you needed more than 8 gpus you can use a larger ray cluster instead, but this is beyond the scope of this example. you can contact logan@thumper for asistance if you need help with this. 
+
+### Tools for Checking Model VRAM requirements: 
+- [can it run](https://huggingface.co/spaces/Vokturz/can-it-run-llm)
+- [LLM-Model-VRAM-Calculator](https://huggingface.co/spaces/NyxKrage/LLM-Model-VRAM-Calculator)
+
+## Wrapping Up
+vLLM may be one of the best open source LLM servers available today,for most applications with multiple users. We hope this blog makes it easier for people to use vLLM in their applications on Akash.  If you have any questions or are looking for some assistance feel free to reach out to logan@thumper.ai  
+
+![WizardLM 2 Meme](imgs/wizardllm2meme.png)
```

Comment:
I like this meme personally, but would suggest removing for this piece. We like to keep the Akash blog a bit more formal.

Reply:
i have removed the meme